### PR TITLE
Add serde to TweakedPublicKey

### PR DESCRIPTION
**Pull request by JeremyRubin**

### CODE REVIEW

File: src/util/schnorr.rs [hunk 1]
```diff
@@ -40,6 +40,8 @@ pub type UntweakedPublicKey = ::XOnlyPublicKey;
 
 /// Tweaked BIP-340 X-coord-only public key
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(transparent))]
```
Comment (dr-orlovsky) [inline src/util/schnorr.rs:44]: Nit: This can be combined in to a single line

**Review by dr-orlovsky — APPROVED** [review]
ACK 204f477a34bede7bc57f8230536b6454c7e1d259

**Review by apoelstra — APPROVED** [review]
ACK 204f477a34bede7bc57f8230536b6454c7e1d259

**Review by sanket1729 — APPROVED** [review]
utACK 204f477a34bede7bc57f8230536b6454c7e1d259

**Pull request merged**